tools/isobuild/linker.js

```diff
@@ -145,32 +150,121 @@ _.extend(Module.prototype, {
     // Prologue
     var chunks = [];
 
+    const result = {
+      // This object will be populated with .source, .servePath,
+      // .sourceMap, and (optionally) .exportsName properties before being
+      // returned from this method in a singleton array.
+      servePath: self.combinedServePath,
+    };
+
     // Emit each file
-    _.each(self.files, function (file) {
-      if (!_.isEmpty(chunks)) {
-        chunks.push("\n\n\n\n\n\n");
+    if (self.useMeteorInstall) {
+      const tree = {};
+
+      _.each(self.files, function (file) {
+        if (file.bare) {
+          // Bare files will be added in between the synchronous require
+          // calls below.
+          return;
+        }
+
+        const parts = file.installPath.split("/");
+        let t = tree;
+        _.each(parts, function (part, i) {
+          const isLastPart = i === parts.length - 1;
+          t = _.has(t, part)
+            ? t[part]
+            : t[part] = isLastPart ? file : {};
+        });
+      });
+
+      let moduleCount = 0;
+
+      function walk(t) {
```

Comment:
I'll add comments, but this `walk` function is similar to `JSON.stringify(tree)`, except the final result (`chunks`) is an array of strings and `SourceNode`s.